canonical-web-and-design/maas-ui
# feat(ui): add tags from the device config form

## Done

- Made add tag form reusable.
- Add the ability to add new tags from the device config form.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the details page for a device.
- Click on the config tab and click 'Edit'.
- Type in a new tag form and click "Create...".
- The modal should appear.
- Fill out the details and submit.
- The new tag name should appear in the input.
- Submit the form and the new tag should appear in the list of tags.

## Fixes

Fixes: canonical-web-and-design/app-tribe#636.

## Screenshot

<img width="832" alt="Screen Shot 2022-04-07 at 12 03 27 pm" src="https://user-images.githubusercontent.com/361637/162117725-880560a4-02df-4c66-b0a1-97202d69e5b3.png">
